### ISSUE TITLE
Relative links to analysis not compatible with COMPASS work directory

### ISSUE THREAD

**Issue by xylar** [MPAS-Dev/MPAS-Model#517]
Several test cases refer to relative paths to analysis that is not linked properly when the test case is set up with the `--work_dir` flag.  These test cases are:
```
 53: -o ocean -c drying_slope -r hybrid -t 1km
 54: -o ocean -c drying_slope -r hybrid -t 250m
 57: -o ocean -c drying_slope -r sigma -t 1km
 58: -o ocean -c drying_slope -r sigma -t 250m
 59: -o ocean -c drying_slope -r zstar -t 1km
 60: -o ocean -c drying_slope -r zstar -t 250m
 131: -o ocean -c surface_waves -r direct -t 1km
 132: -o ocean -c surface_waves -r thickness_source -t 1km
```
As an example from https://github.com/MPAS-Dev/MPAS-Model/blob/ocean/develop/testing_and_setup/compass/ocean/drying_slope/hybrid/1km/config_analysis.xml#L3-L5:
```
  <add_link source="../../../analysis/comparison.py" dest="comparison.py"/>
  <add_link source="../../../analysis/movie.py" dest="movie.py"/>
  <add_link source="../../../analysis/data" dest="data"/>
```

Since `analysis` is not a test case itself (it has no `config_driver.xml`), it does not get copied to the work directory.  These links therefore need to be provided in this way instead:
```
  <add_link source_path="script_configuration_dir" source="analysis/comparison.py" dest="comparison.py"/>
  <add_link source_path="script_configuration_dir" source="analysis/movie.py" dest="movie.py"/>
  <add_link source_path="script_configuration_dir" source="analysis/data" dest="data"/>
```

**Comment by xylar** [comment]
I am working on a PR to fix issues like this and will just include these changes on the `ocean/develop` side but wanted to point them out to see how these fixes can make their way into `ocean/coastal`, particularly in new test cases.

**Comment by pwolfram** [comment]
Thank you @xylar, much appreciated.

**Comment by xylar** [comment]
Addressed by #514
